test/date-string.test.js

```diff
@@ -0,0 +1,72 @@
+// Copyright IBM Corp. 2014,2016. All Rights Reserved.
+// Node module: loopback-datasource-juggler
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+/* global describe,it */
+/* jshint expr:true */
+
+'use strict';
+
+require('should');
+
+var DateString = require('../lib/date-string');
+var fmt = require('util').format;
+var inspect = require('util').inspect;
+var os = require('os');
+
+describe('DateString', function() {
+  describe('constructor', function() {
+    it('should support a valid date string', function() {
+      var theDate = '2015-01-01';
+      var date = new DateString(theDate);
+      date.should.not.eql(null);
+      date.value.should.eql(theDate);
+      date.toString().should.eql(theDate);
+    });
+
+    it('should throw on invalid input', function() {
+      var fn = () => {
+        var date = new DateString('notadate');
+      };
+      fn.should.throw(new Error('Invalid date'));
+
+      fn = () => {
+        var date = new DateString('2015-01-01 25:00:00');
+      };
+      fn.should.throw(new Error('Invalid date'));
+
+      fn = () => {
+        var date = new DateString(20150101);
+      };
+      fn.should.throw(new Error('Input must be a string'));
+    });
+
+    it('should update internal date on set', function() {
+      var date = new DateString('2015-01-01');
+      date.value = '2016-01-01';
+      date.value.should.eql('2016-01-01');
+      var d = new Date('2016-01-01');
+      // The internal date representation should also be updated!
+      date._date.toString().should.eql(d.toString());
+    });
+
+    it('should allow date with time', function() {
+      var theDate = '2015-01-01 02:00:00';
+      var date = new DateString(theDate);
+      date.value.should.eql(theDate);
+      var d = new Date(theDate);
+      date._date.toString().should.eql(d.toString());
```

Comment:
just curious: is there any difference if we compare with `getTime()` instead of `toString()` here?